keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+
+    // Results lists all allocated devices.
     //
     // +listType=atomic
-    ResourceHandles []ResourceHandle
+    Results []RequestAllocationResult
 
-    // This field will get set by the resource driver after it has allocated
-    // the resource to inform the scheduler where it can schedule Pods using
-    // the ResourceClaim.
+    // Setting this field is optional. If unset, the allocated devices are available everywhere.
     //
-    // Setting this field is optional. If null, the resource is available
-    // everywhere.
     // +optional
-    AvailableOnNodes *core.NodeSelector
+    AvailableOnNodes *v1.NodeSelector
 
-    // Shareable determines whether the resource supports more
-    // than one consumer at a time.
+    // Shareable determines whether the claim supports more than one consumer at a time.
+    //
     // +optional
     Shareable bool
 }
 
-// AllocationResultResourceHandlesMaxSize represents the maximum number of
-// entries in allocation.resourceHandles.
-const AllocationResultResourceHandlesMaxSize = 32
+// AllocationResultsMaxSize represents the maximum number of
+// entries in allocation.results.
+const AllocationResultsMaxSize = 32
 
-// ResourceHandle holds opaque resource data for processing by a specific kubelet plugin.
-type ResourceHandle struct {
-    // DriverName specifies the name of the resource driver whose kubelet
-    // plugin should be invoked to process this ResourceHandle's data once it
-    // lands on a node. This may differ from the DriverName set in
-    // ResourceClaimStatus this ResourceHandle is embedded in.
+// RequestAllocationResult contains configuration and the allocation result for
+// one request.
+type RequestAllocationResult struct {
+    // DriverName specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
     DriverName string
 
-    // StructuredData captures the result of the allocation for this
-    // particular driver.
-    StructuredData *StructuredResourceHandle
-}
-
-// StructuredResourceHandle is the in-tree representation of the allocation result.
-type StructuredResourceHandle struct {
-    // VendorClassParameters are the per-claim configuration parameters
-    // from the resource class at the time that the claim was allocated.
-    VendorClassParameters runtime.Object
+    // Config contains all the configuration pieces that apply to the request
+    // and that were meant for the driver which handles these devices.
+    // They get collected during the allocation and stored here
+    // to ensure that they remain available while the claim is allocated.
+    //
+    // Entries are list in the same order as in class.config and claim.config,
+    // with class.config entries first.
+    //
+    // +optional
+    Config []DeviceConfiguration
```

Comment:
I would put this at the end of the struct

Reply:
Okay.